src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/GCRefMapBuilder.cs

```diff
@@ -290,6 +289,26 @@ private void GcScanValueType(TypeDesc type, ArgDestination argDest, int delta, C
             }
         }
 
+        private void FindByRefPointerOffsetsInByRefLikeObject(TypeDesc type, ArgDestination argDest, int delta, CORCOMPILE_GCREFMAP_TOKENS[] frame)
+        {
+            if (type.IsByReferenceOfT || type.IsByRef)
+            {
+                argDest.GcMark(frame, delta, interior: true);
+                return;
+            }
+
+            foreach (FieldDesc field in type.GetFields())
+            {
+                if (!field.IsLiteral && 
+                    !field.IsStatic && 
+                    field.FieldType.IsValueType && 
+                    field.FieldType.IsByRefLike)
```

Comment:
`IsByRefLike` is a subset of `IsValueType`. I would assert `IsValueType` for clarity, but we don't need to check for it in the `if`.

Reply:
Fixed in 3rd commit, thanks for spotting these simplifications.